website/docs/r/waf_rule.html.markdown

```diff
@@ -50,11 +50,11 @@ The following arguments are supported:
 #### Arguments
 
 * `negated` - (Required) Set this to `false` if you want to allow, block, or count requests
-  based on the settings in the specified `ByteMatchSet`, `IPSet`, `SqlInjectionMatchSet`, `XssMatchSet`, or `SizeConstraintSet`.
+  based on the settings in the specified `IPMatch`, `ByteMatch`, `SqlInjectionMatch`, `SizeConstraint` or `XssMatch`.
```

Comment:
I agree the list here may not be very helpful, but the patch here looks rather like a mistake? Because the context here is actually resource, not type, AFAICT.

I think we could either leave it as it was or change it to human-readable names (e.g. Byte Match Set, IP Set, ...) or to Terraform resource names (e.g. `aws_waf_byte_match_set`, `aws_waf_ipset`) possibly with links to docs of those resources.

What do you think?

Reply:
updated with links to terraform resources